src/lib/y2network/hwinfo.rb

```diff
@@ -17,14 +17,69 @@
 # To contact SUSE LLC about this file by physical or electronic mail, you may
 # find current contact information at www.suse.com.
 
+require "yast"
+
 module Y2Network
-  # Physical interface hardware information
+  # Stores useful (from networking POV) items of hwinfo for an interface
+  # FIXME: decide whether it should read hwinfo (on demand or at once) for a network
+  # device and store only necessary info or just parse provided hash
   class Hwinfo
-    # @return [Hash] hwinfo
-    attr_accessor :data
+    attr_reader :hwinfo
+
+    def initialize(name:)
+      # FIXME: store only what's needed.
+      @hwinfo = load_hwinfo(name)
+    end
+
+    # Shortcuts for accessing hwinfo items. Each hwinfo item has own method for reading
+    # its value. There are two exceptions however. First exception is hwinfo["name"] item
+    # which carries device model human friendly description. This item is accessible via
+    # {Hwinfo::description}. Second exception is hwinfo["dev_name"] item which can be read
+    # through {Hwinfo::dev_name} or its alias {Hwinfo::name}.
```

Comment:
For comparison, in the Salt Widgets feature, I have explicitly removed a `name` attribute, because it was unclear whether it referred to a machine readable form (`id`) or a human readable form (`label`).

I will leave this to @jreidinger to judge, because I don't see the big picture here.